addons/sourcemod/scripting/ConnectAnnounce.sp

```diff
@@ -64,12 +80,21 @@ public void OnPluginStart()
 	AutoExecConfig(true);
 }
 
-public void OnPluginEnd()
+//----------------------------------------------------------------------------------------------------
+// Purpose:
+//----------------------------------------------------------------------------------------------------
+public void OnMapEnd()
 {
-	if (g_hDatabase != null)
+	// Clean up on map end just so we can start a fresh connection when we need it later.
+	// Also it is necessary for using SQL_SetCharset
+	if (g_hDatabase)
 		delete g_hDatabase;
+
 	if (g_hDatabase_Hlstatsx != null)
 		delete g_hDatabase_Hlstatsx;
+
+	g_hDatabase = null;
```

Comment:
delete already sets variables to null so not needed:
- https://forums.alliedmods.net/showpost.php?p=2541943&postcount=7